spec/unit/multi/advance_spec.rb

```diff
@@ -122,4 +122,53 @@
       restore
     ].join)
   end
+
+  it "advances progress bars correctly with indeterminate children" do
+    bars = described_class.new("total: :current", output: output)
+
+    bar1 = bars.register("one: :current")
+    bar2 = bars.register("two: :current")
+
+    bar1.advance
+
+    output.rewind
+    expect(output.read).to eq([
+      "\e[1G#{top}total: 1\n",
+      "\e[1G#{bottom}one: 1\n"
+    ].join)
+
+    bar2.advance
+
+    output.rewind
+    expect(output.read).to eq([
+      "\e[1G#{top}total: 1\n",
+      "\e[1G#{bottom}one: 1\n",
+      save,
+      "\e[2A", # up 2 lines
+      "\e[1G#{top}total: 2",
+      restore,
```

Comment:
This spec passes the current behaviour of the multi-output, but once I grok'd what was going on, I was surprised that line 2 does not get rewritten as `"\e[1G#{middle}one: 1",`, specifically changing the `bottom` token to `middle` for line 2. Is this a bug that warrants fixing?

Reply:
When a given registered bar gets advanced it will update only its progress and display in the multibar, no other registered bar will get its display updated. Is this a bug, well it depends on how you look at it. From the perspective of the display, you may have a situation like in the test when two lines have the same prefix character, e.i. bottom, which will look inaccurate. However, there will never be a situation when the progress bar itself displays inaccurate information.

To synch all registered progress bars to display 'accurate' prefix at the initial advance is rather complex. The complexity arises when, for example, you have 10 registered bars running in their own threads. Bars will advance in random order and random times and may finish painting well before the others start. A finished bar cannot be repainted. In practice, I haven't found this to be an issue as once bars start advancing each bar gets correct prefix character. For me, this is a balance between adding more complexity and having initial 'inaccurate display' for one item. 

Having said all that, do not read it as me not wanting to make this work as expected. If you're up for a challenge and wish to investigate, I will be most appreciative. However, I believe this should happen in a separate PR.